app/_src/kubernetes-ingress-controller/references/annotations.md

```diff
@@ -659,3 +662,33 @@ Sets the max retries on a request. For example, setting this annotation to `3`
 will re-send the request up to three times if it encounters a failure, such as
 a timeout.
 {% endif_version %}
+
+{% if_version gte:2.12.x %}
+### konghq.com/rewrite
+
+> Available since controller 2.12
+
+Rewrite a URL path. This annotation is a shorthand method of applying a
+[request-transformer plugin](/hub/kong-inc/request-transformer/)
+with [a `replace.uri` action](/hub/kong-inc/request-transformer/configuration/#config-replace).
+It cannot be combined with a `konghq.com/plugins` annotation that applies a
+request-transformer plugin as such.
+
+The annotation can rebuild URLs using segments captured from a regular
+expression path. A `$n` in the annotation path represents the nth capture group
+in the Ingress rule path, starting from 1. For example, combining an Ingress
+rule with path `/~/api/(.*)/foo/svc_(.*)` and a `konghq.com/rewrite:
+/v$1/svc/$2` would send a request for `/v2/svc/pricing` upstream for an inbound
+request to `/api/2/foo/svc_pricing` (the `/~` prefix instructs Kong to treat
+the path as a regular expression, and isn't used in the actual request).
+
+Annotations apply at the Ingress level and not modify individual rules. As
+such, this annotation should only be used on Ingresses with a single rule, or
+on Ingresses whose rules paths _all_ match the rewrite pattern.
+
+Note that this annotation overrides `strip_path` and Service `path`
```

Comment:
```suggestion
Note that this annotation overrides [konghq.com/strip-path](/kubernetes-ingress-controller/{{page.kong_version}}/references/annotations/#konghqcomstrip-path) and Service `path`
```
WDYT?

Reply:
The rest of the doc usually leaves annotation mentions unlinked, and there's nothing in particular in its longer description that I think we need here, so leaving them as plain codetext.

Reply:
Linking maybe is overkill (I've proposed it, because someone may try to achieve with `konghq.com/rewrite` what is provided with `konghq.com/strip-path`). 

But to be consistent I think it should be changed to `konghq.com/strip-path` there is no annotation `strip_path`

Reply:
Yeah, that was me confusing the annotation and Kong config formats. They now use hyphens and include the prefix.